### ISSUE TITLE
Perspective camera mode fix [experimental]

### ISSUE THREAD

**Issue by SETENTIAdev** [aarthificial/pixelgraphics#14]
Ok, as some users have been having this issue, I thought I'd give it a go and I think a found a solution.

If you have an orthographic camera, all is good:
![GRASS1](https://user-images.githubusercontent.com/591725/228667525-8b990653-dd4c-457a-bc78-c4f5a36e8bef.gif)

Now, if you use perspective, then this weirdness will happen:
![GRASS2](https://user-images.githubusercontent.com/591725/228667528-f27166d1-26b9-4f25-bdf8-2fdb1a817072.gif)

After some fiddling, here's my fix, on perspective mode:
![perspective-fix](https://user-images.githubusercontent.com/591725/228667544-78962f51-6166-435e-9ec9-5ecf6799b60f.gif)

So, what changed? Basically, after dumping some values, I noticed the values for `_originalCamera.projectionMatrix` had a very noticeable difference between two camera modes. Like, in ortho you have tiny values like `0.09`, while in perspective, this number was the same, but with less zeros (ie `0.9`), so I kinda added this zero back, and all is good.

Basically the fix would be replacing this line:

`cmd.SetGlobalVector(ShaderIds.CameraPositionDelta, screenDelta * 0.5f);`

... for this one:
`cmd.SetGlobalVector(ShaderIds.CameraPositionDelta, screenDelta * 0.05f);`

I tested different camera Z positions, FOV values and all seem to be working fine, but needs more testing.

If anyone is happy to further test or have the definitive fix, feel free to contribute. Hope this is useful for someone as it is for me!

**Comment by WatchingFromFar** [comment]
In which file do I have to replace this?

**Comment by SETENTIAdev** [comment]
> In which file do I have to replace this?

Hey, file is `VelocityCamera.cs`, located in: _`Runtime/Universal/`_ folder -- Line `111`

**Comment by Heurazio** [comment]
This is great. Thanks a lot for the info.
